pages/Student/List/index.jsx

```diff
@@ -0,0 +1,341 @@
+import React from "react";
+import { DataGrid, GridToolbarQuickFilter } from "@mui/x-data-grid";
+import { Typography, Box, styled } from "@mui/material";
+
+import MaleIcon from "@mui/icons-material/Male";
+import FemaleIcon from "@mui/icons-material/Female";
+
+import { DashboardLayout } from "@components/Global/Layout";
+
+export const genders = [
+  { name: "male", label: "Male", icon: <MaleIcon /> },
+  { name: "female", label: "Female", icon: <FemaleIcon /> },
+];
+
+const columns = [
+  {
+    field: "id",
+    headerName: "ID",
+    minWidth: 70,
+    flex: 0.5,
+    renderCell: (params) => params.row.id || "N/A",
+  },
+  {
+    field: "name",
+    headerName: "Student Name",
+    minWidth: 150,
+    flex: 1,
+    renderCell: (params) => params.row.name || "N/A",
+  },
+  {
+    field: "parentName",
+    headerName: "Parent Name",
+    minWidth: 150,
+    flex: 1,
+    renderCell: (params) => params.row.parentName || "N/A",
+  },
+  {
+    field: "age",
+    headerName: "Age",
+    type: "number",
+    minWidth: 100,
+    flex: 0.5,
+    align: "left",
+    headerAlign: "left",
+    renderCell: (params) =>
+      params.row.age !== null && params.row.age !== undefined
+        ? params.row.age
+        : "N/A",
+  },
+  {
+    field: "grade",
+    headerName: "Grade",
+    minWidth: 100,
+    flex: 0.5,
+    renderCell: (params) => params.row.grade || "N/A",
+  },
+  {
+    field: "mobile",
+    headerName: "Mobile Number",
+    minWidth: 150,
+    flex: 1,
+    renderCell: (params) => params.row.mobile || "N/A",
+  },
+  {
+    field: "gender",
+    headerName: "Gender",
+    headerAlign: "center",
+    align: "center",
+    minWidth: 100,
+    flex: 0.5,
+    renderCell: (params) => {
+      switch (params.value) {
+        case "male":
+          return <MaleIcon />;
+        case "female":
+          return <FemaleIcon />;
+        default:
+          return "N/A";
+      }
+    },
+  },
+  {
+    field: "subjects",
+    headerName: "Subjects",
+    minWidth: 200,
+    flex: 2,
+    renderCell: (params) =>
+      params.value && params.value.length > 0 ? params.value.join(", ") : "N/A",
+  },
+];
+
+const rows = [
```

Comment:
don't make a file too big to read. move it to another file for better readilbilty